### PR TITLE
Common look and feel to NeMo Collections argument parsing

### PR DESCRIPTION
**Pull request by ericharper**
We add argument parsing similar to PyTorch Lightning's Trainer.add_argparse_args.  This change enables us to have the same arguments across a NeMo collection or Nemo Core in addition to the same arguments from PTL Trainer.  This will help to enforce a common look and feel to NeMo Collections, while also making example scripts cleaner by not having to repeat the same arg parsing every time.

As an example, see the below run script for ASR.  All NeMo example scripts will have the same arguments for training from PTL trainer and then all NeMo example ASR scripts will have common ASR arguments as well like `--asr-model`.

The specific arguments can easily be added to `collections.asr.arguments.py` by maintainers and contributors.

Example scripts will look like:

```
from argparse import ArgumentParser
import pytorch_lightning as pl
from nemo.core.arguments import add_nemo_args # not sure if this will be necessary
from nemo.collections.asr.arguments import add_asr_args
...

def main(args):
...

if __name__ == '__main__':

        parser = ArgumentParser()
        parser = pl.Trainer.add_argparse_args(parser)
        parser = add_nemo_args(parser) 
        parser = add_asr_args(parser)
        parser.add_argument(“—custom_arg_for_example”) 
        args = parser.parse_args()

        main(args)
```

And run scripts will look like:

```
#!/bin/bash

python speech_to_text.py \
        --asr_model "my_bad_quartznet15x5.yaml" \
        --train_dataset "/raid/data/an4/an4/train_manifest.json" \
        --eval_dataset "/raid/data/an4/an4/test_manifest.json" \
        --gpus 4 \
        --distributed_backend "ddp" \
        --max_epochs 1 \
        --fast_dev_run \
```

### CODE REVIEW

**Comment by blisc** [comment]
This definitely feels like the right direction.

I feel like in theory that was what [NeMoArgParser](https://github.com/NVIDIA/NeMo/blob/8fec5b4d740eab8dc5462f82c7118edf00678ecb/nemo/utils/argparse.py#L6) was supposed to achieve but it was never adopted. Do you have any advice on ensuring that this paradigm gets adopted?

**Comment by ekmb** [comment]
> This definitely feels like the right direction.
> 
> I feel like in theory that was what [NeMoArgParser](https://github.com/NVIDIA/NeMo/blob/8fec5b4d740eab8dc5462f82c7118edf00678ecb/nemo/utils/argparse.py#L6) was supposed to achieve but it was never adopted. Do you have any advice on ensuring that this paradigm gets adopted?

@blisc at this to the dev guide and docs?

File: nemo/collections/__init__.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# ! /usr/bin/python
```
Comment (okuchaiev) [inline nemo/collections/__init__.py:1]: I am not sure if we need this file

File: nemo/core/__init__.py [hunk 2]
```diff
@@ -12,5 +12,5 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-import nemo.core.classes as classes
-import nemo.core.neural_types as neural_types
+from nemo.core.classes import *
+from nemo.core.neural_types import *
```
Comment (okuchaiev) [inline nemo/core/__init__.py:16]: let's keep namespace for neural-types related stuff at: "nemo.core.neural_types.*"  - it has many element-type classes that would pollute "nemo.core.*"

File: nemo/collections/__init__.py (hunk 1 above)
Comment (ericharper) [reply nemo/collections/__init__.py:1]: > please fix code style

Done.

Comment (ericharper) [reply nemo/collections/__init__.py]: I removed it.

File: nemo/core/__init__.py (hunk 2 above)
Comment (ericharper) [reply nemo/core/__init__.py]: Updated this so now we have:
```
>>> import nemo.core
>>> nemo.core.
nemo.core.Dataset(        nemo.core.NeuralModule(   nemo.core.common          nemo.core.nemo
nemo.core.FileIO(         nemo.core.Serialization(  nemo.core.dataset         nemo.core.neural_types
nemo.core.Model(          nemo.core.Typing(         nemo.core.modelPT         nemo.core.typecheck(
nemo.core.ModelPT(        nemo.core.classes         nemo.core.module
```

**Comment by ericharper** [comment]
Let me update the NLP Collection since that is in there now.

**Pull request merged**